### PR TITLE
add naming support for a number of operations and update kafka version

### PR DESCRIPTION
**Pull request by yijiezhen**
# Background
We want to be able to name the internal topics created by stateful operations (left-join, left-join-windowed..etc). 

# Changes made
* Upgrade to use 2.7 kafka streams
* Remove Fressian as it conflicts with what we use internally
* Add naming support to a number of operations
* Fixed circleci for parkside

### CODE REVIEW

**Review by k2n — COMMENTED** [review]
Would you replace the badges for those in this repository instead of the original? Please add Parkside as the license holder in addition to FundingCircle at the bottom of README. 
parkside-securities_jackdaw__A_Clojure_library_for_the_Apache_Kafka_distributed_streaming_platform_

**Comment by k2n** [comment]
Please consider bumping up the major version and reset or prepend the changes we made in `CHANGELOG.md`. 

https://softwareengineering.stackexchange.com/questions/142384/forking-an-open-source-project-should-one-create-a-new-changelog-file-should-r

**Review by bren-do — APPROVED** [review]
Happy to see these changes!

File: project.clj [hunk 1]
```diff
@@ -15,13 +15,12 @@
...
-                 [org.apache.kafka/kafka-streams "2.3.1"]
-                 [org.apache.kafka/kafka-streams-test-utils "2.3.1"]
-
+                 [io.confluent/kafka-avro-serializer "6.1.1"]
+                 [org.apache.kafka/kafka-clients "2.7.0"]
+                 [org.apache.kafka/kafka-streams "2.7.0"]
+                 [org.apache.kafka/kafka-streams-test-utils "2.7.0"]
                  [org.clojure/clojure "1.10.1" :scope "provided"]
                  [org.clojure/data.json "0.2.6"]
                  [org.clojure/data.fressian "0.2.1"]
```
Comment (k2n) [inline project.clj:26]: @yijiezhen I see Fressian serde was removed, but this dependency still remains. Is there a reason for it?

Comment (k2n) [inline project.clj:26]: @yijiezhen I see Fressian serde was removed, but this dependency still remains. Is there a reason for it?

**Pull request merged**